include/cute/atom/copy_traits_xe.hpp

```diff
@@ -38,58 +38,147 @@
 
 namespace cute {
 
-namespace detail
-{
-  template<class CopyOp>
-  struct is_transpose : bool_constant<false> {};
+namespace detail {
+
+static constexpr auto subgroup_size = 16;
+
+// ==========  size_of_inst  ==========
+template <class, class = void>
+static constexpr bool has_inst_dtype = false;
+
+template <class T>
+static constexpr bool has_inst_dtype<T, cute::void_t<typename T::inst_dtype>> = true;
+
+template <class T, class dtype, class = void>
+static constexpr auto size_of_inst = sizeof(dtype);
+
+template <class T, class dtype>
+static constexpr auto size_of_inst<T, dtype, std::enable_if_t<has_inst_dtype<T>>> = sizeof(typename T::inst_dtype);
```

Comment:
`has_inst_dtype` isn't needed:
```suggestion
static constexpr auto size_of_inst<T, dtype, std::void_t<typename T::inst_dtype>> = sizeof(typename T::inst_dtype);
```

What is `inst_dtype` exactly and why do only XE_2D_U16x16x16_LD_T and XE_2D_U16x16x8_LD_T have them? And why do we use the size of the trait struct otherwise?